server/src/main/java/org/elasticsearch/discovery/zen/PublishClusterStateAction.java

```diff
@@ -288,7 +288,7 @@ private void sendClusterStateToNode(final ClusterState clusterState, BytesRefere
             //  and not log an error if it arrives after the timeout
             // -> no need to compress, we already compressed the bytes
             TransportRequestOptions options = TransportRequestOptions.builder()
-                .withType(TransportRequestOptions.Type.STATE).withCompress(false).build();
+                .withType(TransportRequestOptions.Type.STATE).build();
```

Comment:
I know this is existing, but I think we can lift this up to a singleton so that we do not create a new instance on every publish to every node.